bundles/org.eclipse.osgi/supplement/src/org/eclipse/osgi/report/resolution/ResolutionReport.java

```diff
@@ -193,4 +197,24 @@ public interface Listener {
 	 * @return a resolution report message.
 	 */
 	String getResolutionReportMessage(Resource resource);
+
+	/**
+	 * @return the number of permutations created to solve the problem or -1 if
+	 *         unknown
+	 * @since 3.20
+	 */
+	int getTotalPermutations();
+
+	/**
+	 * @return the number of processed permutations -1 if unknown
+	 * @since 3.20
+	 */
+	int getProcessedPermutations();
+
+	/**
+	 * @return the number of permutations cause by a use-constraint violation -1 if
+	 *         unknown
+	 * @since 3.20
+	 */
+	int getUsesPermutations();
```

Comment:
So we don't care about other permutation types?  Could we do without this method for now?

Reply:
For now I mostly care about uses permutations, other might become important later on but currently these are the ones I like to cut down as they often are a sign of suboptimal choices of the resolver, e.g. ideally the resolver will never choose a permutation that results in a use constraint violation and there ore this number should go down to zero for a testcase that shows such sub optimal behavior.